Arm/Memory/SeparateAutomation.lean

```diff
@@ -385,9 +388,20 @@ def simpAndIntroDef (name : String) (hdefVal : Expr) : SimpMemM FVarId  := do
     let hdefTy ← inferType hdefVal
 
     /- Simp to gain some more juice out of the defn.. -/
-    let (simpCtx, simprocs) ← LNSymSimpContext
-      (config := { decide := false, failIfUnchanged := false })
-    let (simpResult, _stats) ← simp hdefTy simpCtx simprocs
+    let mut simpTheorems : Array SimpTheorems := #[]
+    for a in #[`minimal_theory, `bitvec_rules] do
+      let some ext ← (getSimpExtension? a)
+        | throwError m!"[simp_mem] Internal error: simp attribute {a} not found!"
+      simpTheorems := simpTheorems.push (← ext.getTheorems)
+
+    -- unfold `state_value.
+    simpTheorems := simpTheorems.push <| ← ({} : SimpTheorems).addDeclToUnfold `state_value
+    let simpCtx : Simp.Context := { 
+      simpTheorems,
+      config := { decide := true, failIfUnchanged := false },
+      congrTheorems := (← Meta.getSimpCongrTheorems)
+    }
```

Comment:
```suggestion
LNSymSimpContext 
   (config := { decide := false, failIfUnchanged := false }) 
   (simp_attrs := #[`minimal_theory, `bitvec_rules])
   (decls_to_unfold := #[state_value])
```

@bollu I believe `LNSymSimpContext` can work here. Fair warning: haven't tried out the suggestion, excuse typos, etc.

Reply:
Also, open to making `LNSymSimpContext` nicer (separate issue).

Reply:
@shigoel I think I would prefer to keep it as-is, because it's hard to see what `LNSymSimpContext` builds unless one goes and reads its code. Since this code is meant to be performant, I would rather have the building of the simp-set inline, so it's easy to audit.

Reply:
Weren't you using `LNSymSimpContext` before though?
https://github.com/leanprover/LNSym/pull/126/commits/dd24b6dd0f2c6d7fd8ab0cde42ccd1e75b0838df

Not a big deal, but I want to make sure I understand what the reasons are.

Reply:
Indeed, I was reusing `LNSymSimpContext` to make sure that if `LNSymSimpContext` got more powerful, I would get the same power for free :) But thinking this through, I found your argument for why this should *not* be  expensive to be compelling. Therefore, I'd prefer to build a custom simp-set that's tuned down, and to add power to it as necessary. 

From an engineering perspective, I'm advocating for [WET / write everything twice] (https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) instead of DRY. 😉

Reply:
Gotcha, though I would note that the particular invocation of `LNSymSimpContext` is equivalent to what you wrote inline.